partner_identification/models/res_partner.py

```diff
@@ -8,12 +8,164 @@
+                ),
+                search=lambda s, *a: s._search_identification(
+                    'SSN', *a
+                ),
+            )
+
+        Args:
+            field_name (str): Name of field to set.
+            category_code (str): Category code of the Identification type.
+        """
+        for record in self:
+            id_number = record.id_numbers.filtered(
+                lambda r: r.category_id.code == category_code
+            )
+            record_len = len(id_number)
+            # Record for category is not existent.
+            if record_len == 0:
+                name = record[field_name]
+                if not name:
+                    # No value to set
+                    continue
+                category = self.env['res.partner.id_category'].search([
+                    ('code', '=', category_code),
+                ])
+                if not category:
+                    category = self.env['res.partner.id_category'].create({
+                        'code': category_code,
+                        'name': category_code,
+                    })
+                self.env['res.partner.id_number'].create({
+                    'partner_id': record.id,
+                    'category_id': category.id,
+                    'name': name,
+                })
+            # There was an identification record singleton found.
+            elif record_len == 1:
+                value = record[field_name]
+                if value:
+                    id_number.name = value
+                else:
+                    id_number.active = False
+            # Guard against writing wrong records.
+            else:
+                raise ValidationError(_(
+                    'This %s has multiple IDs of this type (%s), so a write '
+                    'via the %s field is not possible. In order to fix this, '
+                    'please use the IDs tab.',
+                ) % (
+                    record._name, category_code, field_name,
+                ))
+
+    @api.model
+    def _search_identification(self, category_code, operator, value):
+        """ Search method for an identification field.
+
+        Example:
+
+            .. code-block:: python
+
+            social_security = fields.Char(
+                compute=lambda s: s._compute_identification(
+                    'social_security', 'SSN',
+                ),
+                inverse=lambda s: s._inverse_identification(
+                    'social_security', 'SSN',
+                ),
+                search=lambda s, *a: s._search_identification(
+                    'SSN', *a
+                ),
+            )
+
+        Args:
+            category_code (str): Category code of the Identification type.
+            operator (str): Operator of domain.
+            value (str): Value to search for.
+
+        Returns:
+            list: Domain to search with.
+        """
+        id_numbers = self.env['res.partner.id_number'].search([
```

Comment:
@pankk Avoid this (I know this is on v11 - I'll try to fix it later).

It's not necessary to do a search before returning (two queries instead of one).

´´´python
return [
    ('id_numbers.name', operator, value),
    ('id_numbers.category_id.code', '=', category_code),
]
´´´
It will impact performances.

Reply:
I get #434 when doing it like this. Can we go with @pedrobaeza's suggestion in #436?
![image](https://user-images.githubusercontent.com/3942631/43973943-3ddf948e-9cd9-11e8-8b5b-6b0debda30a6.png)

Reply:
@pankk Yes, I was not aware of that bug. Ok for me